src/video.zig

```diff
@@ -678,6 +678,338 @@ pub const egl = struct {
+    ///
+    /// ## Version
+    /// This function is available since SDL 3.2.0.
+    pub fn getCurrentContext() !gl.Context {
+        const ret = C.SDL_GL_GetCurrentContext();
+        return .{
+            .value = try errors.wrapNull(*C.struct_SDL_GLContextState, ret)
+        };
+    }
+
+    /// Get the currently active OpenGL window.
+    ///
+    /// ## Thread Safety
+    /// This function should only be called on the main thread.
+    ///
+    /// ## Version
+    /// This function is available since SDL 3.2.0.
+    pub fn getCurrentWindow() !Window {
+        const ret = C.SDL_GL_GetCurrentWindow();
+        return Window{
+            .value = try errors.wrapNull(*C.SDL_Window, ret)
+        };
+    }
+
+    /// Get an OpenGL function by name.
+    ///
+    /// ## Return Value
+    /// Returns a pointer to the named OpenGL function. The returned pointer should be cast to the appropriate function signature.
+    ///
+    /// # Remarks
+    /// If the GL library is loaded at runtime with SDL_GL_LoadLibrary(), then all GL functions must be retrieved this way.
+    /// Usually this is used to retrieve function pointers to OpenGL extensions.
+    ///
+    /// There are some quirks to looking up OpenGL functions that require some extra care from the application.
+    /// If you code carefully, you can handle these quirks without any platform-specific code, though:
+    ///
+    /// * On Windows, function pointers are specific to the current GL context; this means you need to have created a GL context and made it current before calling `video.gl.getProcAddress()`. If you recreate your context or create a second context, you should assume that any existing function pointers aren't valid to use with it. This is (currently) a Windows-specific limitation, and in practice lots of drivers don't suffer this limitation, but it is still the way the wgl API is documented to work and you should expect crashes if you don't respect it. Store a copy of the function pointers that comes and goes with context lifespan.
+    /// * On X11, function pointers returned by this function are valid for any context, and can even be looked up before a context is created at all. This means that, for at least some common OpenGL implementations, if you look up a function that doesn't exist, you'll get a non-NULL result that is NOT safe to call. You must always make sure the function is actually available for a given GL context before calling it, by checking for the existence of the appropriate extension with `video.gl.extensionSupported()`, or verifying that the version of OpenGL you're using offers the function as core functionality.
+    /// * Some OpenGL drivers, on all platforms, will return NULL if a function isn't supported, but you can't count on this behavior. Check for extensions you use, and if you get a NULL anyway, act as if that extension wasn't available. This is probably a bug in the driver, but you can code defensively for this scenario anyhow.
```

Comment:
When NULL is referred to, we should try using `null` when possible as it applies directly to zig that way.